tools/codegen/dest/lazy_ir.py

```diff
@@ -355,3 +358,75 @@ def __call__(self, f: NativeFunction) -> List[str]:
             return ["\n".join([f"{shape_sig.shape_decl};"])]
         else:
             return []
+
+
+@dataclass
+class LazyIrBuilder:
+    schema: NonNativeLazyIrSchema
+
+    def __post_init__(self) -> None:
+        params = [f"const Value& {i}" for i in self.schema.inputs]
+        params.extend(
+            f"const {arg.lazy_type.cpp_type()}& {arg.name}" +
+            (" = c10::nullopt" if isinstance(arg.lazy_type, OptionalCType) else "")
+            for arg in self.schema.scalar_args
+        )
+        self.params = ", ".join(params)
+        args = self.schema.inputs.copy() + [arg.name for arg in self.schema.scalar_args]
+        self.args = ", ".join(args)
+
+        self.signature = f"NodePtr Make{self.schema.node_name}({self.params})"
+
+    def virtual_method(self) -> str:
+        return f"virtual {self.signature} const = 0;"
+
+    def override_method(self) -> str:
+        return (
+            f"{self.signature} const override "
+            f"{{ return MakeNode<{self.schema.node_name}>({self.args}); }}"
+        )
+
+    def make_function(self) -> str:
+        return dedent(
+            f"""
+            static inline {self.signature} {{
+                static IrBuilder* builder = getBackend()->GetIrBuilder();
+                return builder->Make{self.schema.node_name}({self.args});
+            }}
+            """
+        ).strip()
+
+
+def generate_lazy_ir_node_functions(non_native: Dict[str, Any]) -> str:
```

Comment:
I like the style of these templates and thanks for using dedent, i've been meaning to do that for the rest of the codegen!